test/test_unary_ufuncs.py

```diff
@@ -33,6 +273,149 @@ def test_unsupported_dtypes(self, device, dtype, op):
         with self.assertRaises(RuntimeError):
             op(t)
 
+    @dtypes(*floating_types_and(torch.bfloat16, torch.half))
+    @ops((_fn for _fn in unary_ufuncs if _fn.domain is not None))
+    def test_float_domains(self, device, dtype, op):
+        if not op.supports_dtype(dtype, torch.device(device).type):
+            raise unittest.SkipTest('unsupported dtype')
+
+        eps = (1e-1, 1, 2, 10, 20, 50, 100)
+
+        # Adds smaller values except on bfloat16, since these small differences
+        #   are often too small to represent in bfloat16
+        if dtype is not torch.bfloat16:
+            eps = eps + (1e-5, 1e-3)
+
+        for base, _fn in ((op.domain[0], operator.sub), (op.domain[1], operator.add)):
+            for epsilon in eps:
+                v = _fn(base, epsilon)
+                t = torch.tensor(v, device=device, dtype=dtype)
+                result = op(t)
+                self.assertEqual(result.item(), float('nan'),
+                                 msg="input:{0}".format(t.item()))
+
+    # Tests that fn == method == inplace == jit on a simple single tensor input
+    @ops(unary_ufuncs)
+    def test_variant_consistency(self, device, dtype, op):
+        def _fn(t):
+            return op(t)
+
+        t = _make_tensor((5, 5), device, dtype, domain=op.domain)
+        expected = op(t)
+
+        for alt in (op.get_method(), op.get_inplace(), torch.jit.script(_fn)):
+            if alt is None:
+                continue
+
+            actual = alt(t.clone())
+            self.assertEqual(actual, expected)
+
+    # Tests that the function and its (array-accepting) reference produce the same
+    #   values on a range of tensors, including empty tensors, scalar tensors,
+    #   1D tensors and a large 2D tensor with interesting and extremal values
+    #   and discontiguities.
+    @suppress_warnings
+    @ops(unary_ufuncs)
+    def test_reference_numerics(self, device, dtype, op):
```

Comment:
print the input that failed

Reply:
Unfortunately this would be very complicated with our current test architecture. I think it's going to have to wait for something like torch.testing.compare_tensors().

Reply:
At the very least, it would probably be helpful to report which tensor input index was the problem (even if it's not the actual input.)  You could use a subTest for this?

Reply:
I don't think pytest supports unittest.subtest - is there another subtest you were thinking of?

We do report the index in the output, though, which, in this case, is equivalent to the index in the input.

Reply:
TestCase.subTest

Reply:
I added a descriptive error message when the tensors are printable which should cover most of the interesting cases.